geopandas/tools/sjoin.py

```diff
@@ -90,12 +90,27 @@ def sjoin(
     # and store references to the original indices, to be reaffixed later.
     # GH 352
     left_df = left_df.copy(deep=True)
-    left_index_name = left_df.index.name
-    left_df.index = left_df.index.rename(index_left)
+    try:
+        left_index_name = left_df.index.name
+        left_df.index = left_df.index.rename(index_left)
+    except TypeError:
+        index_left = [
+            "index_%s" % lsuffix + str(l) for l, ix in enumerate(left_df.index.names)
```

Comment:
Not necessarily for this PR, but now I start wondering why we actually rename those indexes instead of keeping the original names ..)

Reply:
I was wondering about the same thing. We probably don't, but I wanted to keep this PR minimal bugfix.

Reply:
> I was wondering about the same thing. We probably don't, but I wanted to keep this PR minimal bugfix.

Yes, sure! Will see if I can get to a general clean-up (there is an alternative implementation in the cython branch, maybe I can dig that up)